lib/protobuf/field/base_field.rb

```diff
@@ -122,10 +124,10 @@ def required?
 
       # FIXME: need to cleanup (rename) this warthog of a method.
       def set(message_instance, bytes)
-        return message_instance.__send__(setter, decode(bytes)) unless repeated?
-        return message_instance.__send__(getter) << decode(bytes) unless packed?
+        return message_instance[name] = decode(bytes) unless repeated?
```

Comment:
Does this still flow through the setter? I'm exactly sure why we needed to be calling the setter here, but I want to make sure this isn't unintentionally altering behavior.